libsolidity/analysis/TypeChecker.cpp

```diff
@@ -737,6 +737,46 @@ void TypeChecker::endVisit(ExpressionStatement const& _statement)
 			typeError(_statement.expression().location(), "Invalid integer constant.");
 }
 
+void TypeChecker::endVisit(Conditional const& _conditional)
```

Comment:
Can you use `visit` for this? If yes, then some checks can be shortened to just
`exectType(_conditional.condition(), BoolType());`